src/jetson/main.py

```diff
@@ -122,9 +140,67 @@ def detect(self,
                     kp_y = detections[i, 4 + k * 2 + 1] * img.shape[0]
 
                 bboxes.append((xmin, ymin, xmax, ymax))
+
             return bboxes
 
 
+        elif (self.model_name == 'retinaface'):
+            def postprocess(boxes, conf):
+                """
+                Performs all the postprocessing such as scaling box coordinates
+                to match the size of input image, and discarding all boxes and confidence
+                scores below a detection threshold
+                Args:
+                    boxes- Box coordinates
+                    conf - confidence scores
+
+                Returns boxes and confidence scores that are above confidence threshold
+                """
+                scale = torch.Tensor([self.image_shape[1], self.image_shape[0], self.image_shape[1], self.image_shape[0]])
+                boxes = (boxes * scale / self.resize).numpy()
+                scores = conf.squeeze(0).data.cpu().numpy()[:, 1]
+
+                # ignore low scores
+                inds = np.where(scores > self.detection_threshold)[0]
+                boxes = boxes[inds]
+                scores = scores[inds]
+
+                return boxes, scores
+
+            def do_nms(boxes):
+                """
+                Performs non-max suppression to remove boxes that have high intersection
+                Args:
+                    boxes - face coordinates
+
+                Returns detections that are above a certain IOU threshold
+                """
+                dets = np.hstack((boxes, scores[:, np.newaxis])).astype(np.float32, copy=False)
+                keep = nms_numpy(dets, infer_params['nms_thresh'])
+                dets = dets[keep, :]
+
+                return dets
```

Comment:
I'm not a fan of these inline functions. They aren't actually solving the initial problem of readability right now and they combine parameters with class / function variables which makes it especially hard to read.

The functions themselves look good. they just need a new location